src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,124 @@
+import React, {
+  useMemo,
+  useState,
+  useRef,
+  useCallback,
+  useEffect,
+} from 'react';
+import debounce from 'lodash.debounce';
+import { peopleFromServer } from '../data/people';
+import { Person } from '../types/Person';
+
+type Props = {
+  onSelected: (person: Person | null) => void;
+  delay?: number;
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  onSelected = () => {},
+  delay = 300,
+}) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isListVisible, setIsListVisible] = useState(false);
+
+  const dropdownRef = useRef<HTMLDivElement | null>(null);
+
+  const applyQuery = useCallback(
```

Comment:
it's a bit uncommon, but you could useMoemo here
or just disable the lint rule for this line

just FYI